src/main/kotlin/org/jitsi/nlj/transform/node/outgoing/RetransmissionSender.kt

```diff
@@ -51,23 +54,30 @@ class RetransmissionSender(
     private var numRetransmittedRtxPackets = 0
     private var numRetransmittedPlainPackets = 0
 
+    private val dummyMap = ObservableMap<Byte, PayloadType>()
+
     init {
-        streamInformationStore.onRtpPayloadTypesChanged { currentRtpPayloadTypes ->
-            if (currentRtpPayloadTypes.isEmpty()) {
-                associatedPayloadTypes.clear()
-            } else {
-                currentRtpPayloadTypes.values.filterIsInstance<RtxPayloadType>()
-                    .map { rtxPayloadType ->
-                        rtxPayloadType.associatedPayloadType?.let { associatedPayloadType ->
-                            associatedPayloadTypes[rtxPayloadType.pt.toInt()] = associatedPayloadType
-                            logger.cdebug { "Associating RTX payload type ${rtxPayloadType.pt.toInt()} " +
-                                "with primary $associatedPayloadType" }
-                        } ?: run {
-                            logger.cerror { "Unable to parse RTX associated payload type from payload " +
-                                "type $rtxPayloadType" }
-                        }
-                    }
+        object : MapEventValueFilterHandler<Byte, PayloadType>({ it is RtxPayloadType }) {
+            override fun entryAdded(key: Byte, value: PayloadType) =
+                setRtxPayloadTypeAssociation(value as RtxPayloadType)
+
+            override fun entryUpdated(key: Byte, newValue: PayloadType) =
+                setRtxPayloadTypeAssociation(newValue as RtxPayloadType)
+
+            override fun entryRemoved(key: Byte) {
+                associatedPayloadTypes.remove(key.toInt())
             }
+        }.apply { streamInformationStore.onRtpPayloadTypeEvent(this.mapEventHandler) }
+    }
+
+    private fun setRtxPayloadTypeAssociation(rtxPayloadType: RtxPayloadType) {
```

Comment:
Now I see that we have this code in two places (here, and in `RtxHandler`). I would move this to the stream information store. It would maintain the `RTX_PT -> APT` map, and doesn't have to fire any events.

Reply:
i didn't do this because then it would have to be a central, concurrent map and i worried about the overhead, since we access these for every packet (i suppose it's likely negligible, but we do have to worry about concurrency there as they do execute in different thread contexts)